### PR TITLE
Rename `Rmtmps` module to `RmUnused`

### PR DESCRIPTION
**Pull request by just-max**
Also change `Rmtmps.removeUnusedTemps` to `RmUnused.removeUnused`.

### CODE REVIEW

File: src/rmUnused.ml [hunk 1]
```diff
@@ -36,8 +36,8 @@
 
  *)
 
-(* rmtmps.ml *)
```
Comment (michael-schwarz) [inline src/rmUnused.ml:39]: Maybe we can add some documentation comment to this module to indicate that this is a rename of `RmTemps`?

File: src/main.ml [hunk 2]
```diff
@@ -61,9 +61,9 @@ let parseOneFile (fname: string) : C.file =
 
   if (not (Feature.enabled "epicenter")) then (
     (* sm: remove unused temps to cut down on gcc warnings  *)
```
Comment (michael-schwarz) [inline src/main.ml]: Is this comment actually up-to-date? Or does it just remove all unused code?

File: test/testcil.pl [hunk 3]
```diff
@@ -469,8 +469,8 @@ sub addToGroup {
                         "Notbug. Should fail since we don't pad argument lists");
 addTest("test/restrict EXTRAARGS=-std=c9x _GNUCC=1");
 addTest("test/restrict1 _GNUCC=1");
-addTest("testrun/rmtmps1 ");
-addTest("testrun/rmtmps2 _GNUCC=1");
+addTest("testrun/rmUnused1 ");
+addTest("testrun/rmUnused2 _GNUCC=1");
```
Comment (michael-schwarz) [inline test/testcil.pl:473]: Why does the name of the test matter here? That seems odd?

**Review by michael-schwarz — APPROVED** [review]

File: test/testcil.pl (hunk 3 above)
Comment (just-max) [reply test/testcil.pl:473]: I renamed the C file that contains the test: [test/small1/rmtmps1.c → test/small1/rmUnused1.c](https://github.com/goblint/cil/pull/135/files#diff-0723c1182ff96b92335bc41cd0ea94c133f2a9382b13b9b5bb0661940a45ccdf). Not strictly necessary, but keeps things consistent.

(In my first commit, I forgot to change the addTest call, hence the second commit.)

File: src/main.ml (hunk 2 above)
Comment (just-max) [reply src/main.ml]: Looks like it removes all unused code.

Comment (michael-schwarz) [reply src/main.ml]: Maybe we can also update the text here then?

**Review by sim642 — APPROVED** [review]

File: src/main.ml (hunk 2 above)
Comment (michael-schwarz) [reply src/main.ml]: Once we update this text, I think it's good to merge!

**Comment by just-max** [comment]
Should be ready now!

File: src/main.ml (hunk 2 above)
Comment (sim642) [reply src/main.ml]: c2e97b91e380f8a19ea6343e2179e660f60c8526 removed the entire epicenter feature check, not just updated the comment.

Comment (michael-schwarz) [reply src/main.ml]: The epicenter feature has long been removed from our fork, and the upstream version is no longer compatible with the current code base without rewriting.

**Pull request merged**